src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -362,6 +373,10 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
       }
     }
 
+    if (!query_slots_to_reset.empty()) {
+      timer_query_pool_->ResetQuerySlots(device, query_slots_to_reset);
```

Comment:
This is a bit problematic. We would set the slot directly as available again. However, the slot is wired into the command buffer, being submitted. Thus, it will write into this slot index and our results might be wrong, if the slot is being picked up again.
However, this will be addressed by my change with the re-use of command buffers. Could you add a TODO for now, you can reference the b/181098020.

Reply:
> However, the slot is wired into the command buffer, being submitted.

It's possible I misunderstand something, but we reset the slot in the command buffer above (the new "if" above). Doesn't that mean that we don't get the timestamp there?

Reply:
Ah, I get it, it's in the command buffer we send to the GPU. Yes, that's a problem.

Reply:
So in the if above:
end is nullopt, begin has lets say slot i. So in the command buffer, there is a "CmdWriteTimestamp(i)".
This command buffer is submitted exactly now, so it will be executed.
What we are doing here, is reset the slot on the hardware (done immediately, so before the write timestamp command was executed) and mark the slot as free again. Further, above, we ensure, that we will not read the slot in the present.

Now if right after this code, another command buffer (or marker or so) begins, and thus requests a slot, the pool may reserve "i" for it (as it was resetted).
Now we have two command buffers writing into the same slot. Thus undefined what the result will be. Still that case is very very unlikely. Especially, as the new command buffer must be created, submitted and finished with the execution, BEFORE, the one from the current submission executes, but still, it is possible.

Reply:
Added a TODO and comment as you suggested. I think it's not that much of a problem right now as this only happens in the case when we run out of slots, in which case we typically don't get any timeslices anyway. But should be addressed at some point.

Reply:
But as I said, this is solvable with the exact same logic, I am currently implementing. We would mark those slots as "done reading", and just wait until their command buffers get "reset" to free the slot again.